src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,178 @@
+        for (Map.Entry<K, V> entry : map.entrySet()) {
+            put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    public boolean containsKey(K key) {
+        Node<K, V> node = table[getHash(key)];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return true;
+            }
+            node = node.next;
+        }
+        return false;
+    }
+
+    public boolean containsValue(V value) {
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                if (Objects.equals(value, node.value)) {
+                    return true;
+                }
+                node = node.next;
+            }
+        }
+        return false;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getHash(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    /*
+    wasn't tested
+     */
+    public V remove(K key) {
+        Node<K, V> node = table[getHash(key)];
+        if (node == null) {
+            return null;
+        }
+        if (Objects.equals(key, node.key)) {
+            table[getHash(key)] = node.next;
+            return node.value;
+        }
+        Node<K, V> next = node.next;
+        while (next != null) {
+            if (Objects.equals(next.key, key)) {
+                node.next = next.next;
+                return next.value;
+            }
+            node = node.next;
+            next = next.next;
+        }
         return null;
     }
 
+    /*
+    removes false if key wasn't found in the HashMap
+    true if such a key was found, thus removed
+     */
+    public boolean remove(K key, V value) {
+        if (remove(key) == null) {
+            return false;
+        }
+        return true;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    public List<V> getValues() {
+        List<V> list = new ArrayList<>();
+        if (size == 0) {
+            return list;
+        }
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                list.add(node.value);
+                node = node.next;
+            }
+        }
+        return list;
+    }
+
+    private void resize() {
+        capacity = capacity * GROWTH_FACTOR;
+        threshold = threshold * GROWTH_FACTOR;
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[capacity];
+        for (Node<K, V> node : table) { //Transfer
+            while (node != null) {
+                putToTable(node.key, node.value, newTable, true);
+                node = node.next;
+            }
+        }
+        table = newTable;
+    }
+
+    private void putToTable(K key, V value, Node<K, V>[] table, boolean resize) {
```

Comment:
`boolean resize` is redundant because you can do `size--;` in method `resize()` and you will don't need operations like this `size = resize ? size : size + 1;`